File: yts_project_vueai/src/api/modules/hotStampingPatternAreaCompatibility.ts
import axios from 'axios'

const API_BASE_URL = '/api'

// 定义接口类型
export interface HotStampingPatternAreaCompatibility {
  id?: number
  productName: string
  hotStampingPatternxAreaId: number
  compatibilityStatus: string
  paperType?: string
  optionName?: string
  patternType?: string
  createdAt?: string
  updatedAt?: string
}

export interface ApiResponse<T> {
  success: boolean
  data: T
  message?: string
}

// API 调用函数
export const hotStampingPatternAreaCompatibilityApi = {
  // 获取所有兼容性配置列表
  getAllCompatibility: async (): Promise<HotStampingPatternAreaCompatibility[]> => {
    const response = await axios.get(`${API_BASE_URL}/api/hot-stamping-pattern-area-compatibility/all`)
    return response.data
  },
  
  // 根据图案区域ID获取兼容性配置列表
  getCompatibilityByPatternAreaId: async (patternAreaId: number): Promise<HotStampingPatternAreaCompatibility[]> => {
    const response = await axios.get(`${API_BASE_URL}/api/hot-stamping-pattern-area-compatibility/pattern-area/${patternAreaId}`)
    return response.data
  },

  // 根据ID获取兼容性配置
  getCompatibilityById: async (id: number): Promise<HotStampingPatternAreaCompatibility> => {
    const response = await axios.get(`${API_BASE_URL}/api/hot-stamping-pattern-area-compatibility/${id}`)
    return response.data
  },

  // 创建兼容性配置
  createCompatibility: async (compatibility: Omit<HotStampingPatternAreaCompatibility, 'id'>): Promise<HotStampingPatternAreaCompatibility> => {
    const response = await axios.post(`${API_BASE_URL}/api/hot-stamping-pattern-area-compatibility`, compatibility)
    return response.data
  },

  // 更新兼容性配置
  updateCompatibility: async (id: number, compatibility: Partial<HotStampingPatternAreaCompatibility>): Promise<HotStampingPatternAreaCompatibility> => {
    const response = await axios.put(`${API_BASE_URL}/api/hot-stamping-pattern-area-compatibility/${id}`, compatibility)
    return response.data
  },

  // 删除兼容性配置
  deleteCompatibility: async (id: number): Promise<void> => {
    await axios.delete(`${API_BASE_URL}/api/hot-stamping-pattern-area-compatibility/${id}`)
  },

  // 根据图案区域ID删除所有兼容性配置
  deleteCompatibilityByPatternAreaId: async (patternAreaId: number): Promise<void> => {
    await axios.delete(`${API_BASE_URL}/api/hot-stamping-pattern-area-compatibility/pattern-area/${patternAreaId}`)
  },

  // 获取所有可用的燙金紙系列
  getAllProductNames: async (): Promise<string[]> => {
    const response = await axios.get(`${API_BASE_URL}/api/hot-stamping-pattern-area-compatibility/product-names`)
    return response.data
  }
}

File: yts_project/src/main/java/com/it/yts_project/model/HotStampingPatternAreaCompatibility.java
package com.it.yts_project.model;

import com.alibaba.excel.annotation.ExcelIgnore;
import com.alibaba.excel.annotation.ExcelProperty;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

/**
 * 烫金图案区域兼容性表实体类
 * 对应数据库表: hot_stamping_patternx_area_compatibility
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class HotStampingPatternAreaCompatibility {
    
    @ExcelIgnore
    private Integer id;
    
    /**
     * 燙金紙系列
     */
    @ExcelProperty("燙金紙系列")
    private String productName;
    
    @ExcelIgnore
    private Integer hotStampingPatternxAreaId;
    
    /**
     * 兼容性状态
     * V: 适用, X: 不适用, NA: 不确定, ▷: 需要打底处理
     */
    @ExcelProperty("兼容性狀態(V:适用, X:不适用, NA:不确定, ▷:需要打底处理)")
    private String compatibilityStatus;
    
    /**
     * 纸张类型
     */
    @ExcelProperty("燙金紙性能類型")
    private String paperType;
    
    @ExcelIgnore
    private LocalDateTime createdAt;
    
    @ExcelIgnore
    private LocalDateTime updatedAt;
    
    // 关联查询字段（用于导出显示）
    @ExcelProperty("圖案类型")
    private String patternType;
    
    // 关联查询字段（用于显示图案区域名称）
    @ExcelIgnore
    private String optionName;
}
